src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -175,28 +157,42 @@ void ComponentFiller::addVariables(Optimisation::LinearProblemApi::ILinearProble
     {
         const auto& lb = evaluator.dispatch(variable.LowerBound().RootNode());
         const auto& ub = evaluator.dispatch(variable.UpperBound().RootNode());
+        constexpr int numberOfScenarios = 1; // FIXME ONE SCENARIO ONLY
         if (variable.isTimeDependent())
         {
+            const FullKey key(component_.Id(),
+                              variable.Id(),
+                              ctx.getLastTimeStep() - ctx.getFirstTimeStep() + 1,
```

Comment:
well, in the previous code it was possible to have a first time step other than 0. In think this could be a source of malicious bug

Reply:
i mean first and last ts were actually used ; (t_0 = 3, tf = 5)  ≠ (t_0 = 0, tf = 2)